openapi/openapi/src/test/scala/endpoints4s/openapi/ReferencedSchemaTest.scala

```diff
@@ -196,130 +257,152 @@ class ReferencedSchemaTest extends AnyWordSpec with Matchers {
+        |        "discriminator": {
+        |          "propertyName": "storageType",
+        |          "mapping": {
+        |            "Library": "#/components/schemas/endpoints4s.openapi.ReferencedSchemaTest.Storage.Library",
+        |            "Online": "#/components/schemas/endpoints4s.openapi.ReferencedSchemaTest.Storage.Online"
         |          }
         |        }
         |      },
-        |      "endpoints.Errors" : {
-        |        "type" : "array",
-        |        "items" : {
-        |          "type" : "string"
-        |        }
-        |      },
-        |      "endpoints4s.openapi.ReferencedSchemaTest.Storage.Library" : {
-        |        "type" : "object",
-        |        "properties" : {
-        |          "storageType" : {
-        |            "type" : "string",
-        |            "enum" : ["Library"],
-        |            "example" : "Library"
+        |      "endpoints4s.openapi.ReferencedSchemaTest.Storage.Library": {
+        |        "type": "object",
+        |        "properties": {
+        |          "storageType": {
+        |            "example": "Library",
+        |            "type": "string",
+        |            "enum": [
+        |              "Library"
+        |            ]
         |          },
-        |          "room" : {
-        |            "type" : "string"
+        |          "room": {
+        |            "type": "string"
         |          },
-        |          "shelf" : {
-        |            "type" : "integer",
-        |            "format" : "int32"
+        |          "shelf": {
+        |            "type": "integer",
+        |            "format": "int32"
         |          }
         |        },
-        |        "required" : [
+        |        "required": [
         |          "storageType",
         |          "room",
         |          "shelf"
         |        ]
+        |      },
+        |      "Color": {
+        |        "example": "Blue",
+        |        "type": "string",
+        |        "enum": [
+        |          "Red",
+        |          "Blue"
+        |        ]
+        |      },
+        |      "endpoints.Errors": {
+        |        "type": "array",
+        |        "items": {
+        |          "type": "string"
+        |        }
+        |      },
+        |      "endpoints4s.openapi.ReferencedSchemaTest.Book": {
+        |        "example": {
+        |          "id": "2cb38bf2-ecbf-401c-be37-9ec325ae91a0",
+        |          "title": "Essential Scala",
+        |          "author": {
+        |            "name": "Noel Welsh and Dave Gurnell"
+        |          },
+        |          "isbnCodes": [
+        |
+        |          ],
+        |          "storage": {
+        |            "link": "https://underscore.io/books/essential-scala/",
+        |            "storageType": "Online"
+        |          }
+        |        },
```

Comment:
The example is still here. I believe we should remove it from here. Probably, there is something wrong with the way we handle “named” schemas.

Reply:
Yes, you're probably right that we should remove it, but I'm yet to find out how. With this added feature of silbing values, do we want to leave out examples etc. in the `components/schemas` object altogether or is there still a use for this?

Reply:
In my it is fine to have examples in the components but here the issue is that the example is attached to the wrong model.

I believe the issue is that calling named on a schema returns a Schema. Maybe it should return a different type. But that would lead to a lot of complications...

Reply:
Looking more closely now, I see that you're right that it is attached to the wrong model. In fact, the behavior of the referenced schema example hasn't changed (yet). With this new code, the other models just have added schemas. So we'll either need to prevent the example from being added to `components/schemas` or make sure that it contains the expected example.

In this case first example is added to the generic `Book` schema in this line: https://github.com/endpoints4s/endpoints4s/pull/1020/files#diff-8eee4ea78e0bdeb38dd981988cae59c64e82196da2d576610a4cfcc5f62b1629R77

As a a user I would expect this example to be added to each endpoint that uses the `Book` schema, except if I override it with another example on a specific schema, like on this line: https://github.com/endpoints4s/endpoints4s/pull/1020/files#diff-8eee4ea78e0bdeb38dd981988cae59c64e82196da2d576610a4cfcc5f62b1629R99

Is this a correct assumption? If so, I can see two ways to achieve this in the documentation JSON: either leave examples out of `components/schemas` entirely and do everything in the specific schemas or make sure that the right examples are present in the referenced schemas. I can imagine that the former solution results in more verbose JSON, but is less complex to implement. I'll try some things and see what happens.

Reply:
I still have some difficulty in understanding the purpose of `original` and how it's set. If I search for usages of constructors with `original` and `withOriginal` I cannot find any, yet `original` is set for all three `Reference` schemas (two `Book` and one `Color`). Do you have an idea how `original` is set via `withExample`/`withExampleJsonSchema`?